js/authentication.md

```diff
@@ -564,6 +564,115 @@ export default class App extends React.Component {
 }
 ```
 
+#### Federated with Auth0
+
+You can use `Auth0` as one of the providers of your Cognito Federated Identity Pool. This will allow users authenticated via Auth0 have access to your AWS resources.
+
+Step 1. [how to integrate Auth0 with Cognito Federated Identity Pool](https://auth0.com/docs/integrations/integrating-auth0-amazon-cognito-mobile-apps)
+
+Step 2. Login with `Auth0`, then use the id token returned to get AWS credentials from Cognito Federated Identity Pools using the `Auth.federatedSignIn` method:
+```js
+const { idToken, domain, expiresIn, user } = getFromAuth0();
+
+Auth.federatedSignIn({
+    domain, // The Auth0 Domain,
+    {
+        token: idToken // The id token from Auth0
+        expires_at: expiresIn * 1000 + new Date().getTime() // the expiration timestamp
+    },
+    user // the user object, e.x. { name: username, email: email }
+}).then(cred => {
+    console.log(cred);
+});
+```
+
+Step 3. Now you can get the current user and current Credentials:
+```js
+Auth.currentAuthenticatedUser().then(user => console.log(user));
+Auth.currentCredentials().then(creds => console.log(creds));
+
+// Unfortunately Auth.currentSession() doesn't support federated identities for now. Please store the auth0 session info manually.
```

Comment:
Change to:
// Auth.currentSession() does not currently support federated identities. Please store the auth0 session info manually. <-- Explain a bit how they do this or link to how to do this.